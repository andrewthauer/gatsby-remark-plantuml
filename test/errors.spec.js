const testRemarkPlugin = require(`./test-remark-plugin`)
const { PlantUmlError } = require(`../index`)

describe(`Error Handling`, () => {
  beforeEach(() => {
    jest.resetModules()
  })

  it(`reports line numbers`, async () => {
    const code = `
\`\`\`plantuml
@startuml
error here // first line is #1 after @startuml
@enduml
\`\`\`
`

    await testRemarkPlugin.testPlugin({
      code,
      reporter: {
        error: [
          [
            `Could not generate plantuml diagram: Syntax Error at line 1  `,
            expect.toMatchPlantUmlError(
              new PlantUmlError(`Syntax Error`, 1, ``)
            ),
          ],
        ],
      },
    })
  })
})
