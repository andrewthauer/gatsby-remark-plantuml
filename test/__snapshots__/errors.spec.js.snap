// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Handling reports line numbers 1`] = `
Object {
  "children": Array [
    Object {
      "children": undefined,
      "lang": undefined,
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 6,
          "offset": 81,
        },
        "indent": Array [
          1,
          1,
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 2,
          "offset": 1,
        },
      },
      "type": "html",
      "value": "<p><b><span style=\\"color:red\\">PlantUmlError</span></b></p>
<pre>
    0 @startuml
    1 error here // first line is #1 after @startuml
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Syntax Error at line 1 
    2 @enduml
</pre>
",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 1,
      "line": 7,
      "offset": 82,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;
